code/modules/vehicles/tank.dm

```diff
@@ -0,0 +1,577 @@
+/*
+tankS BY KMC
+THIS IS LIKE REGULAR CM tank BUT IT'S LESS SHIT
+WHOEVER MADE CM TANKS: YOU ARE A BAD CODER!!!!!
+*/
+
+//generic component stuff
+
+/obj/item/tank_weapon //These are really just proof of concept weapons. You'll probably want to adapt them to use the hardpoint crap that CM has.
+	name = "TGS 4 main tank cannon"
+	desc = "A gun that works about 50% of the time, but at least it's open source! It fires tank shells."
+	icon = 'icons/obj/hardpoint_modules.dmi'
+	icon_state = "glauncher"
+	var/obj/item/ammo_magazine/ammo
+	var/obj/vehicle/tank/owner
+	var/list/fire_sounds = list('sound/weapons/tank_cannon_fire1.ogg', 'sound/weapons/tank_cannon_fire2.ogg')
+	var/default_ammo = /obj/item/ammo_magazine/tank/ltb_cannon //What do we start with?
+	var/list/accepted_ammo = list(/obj/item/ammo_magazine/tank/tank_glauncher) //Alternative ammo types that we'll accept. The default ammo is in this by default
+	var/fire_delay
+	var/cooldown = 6 SECONDS //6 second weapons cooldown
+	var/lastfired = 0 //When were we last shot?
+	var/range_safety_check = 3 //Stops marines form shooting their own tank. If your gun is geared for explosives, 3 tiles is good. Minigun and APC cannon don't have this.
+
+/obj/item/tank_weapon/Initialize()
+	. = ..()
+	ammo = new default_ammo(src)
+	accepted_ammo += default_ammo
+
+/obj/item/tank_weapon/secondary_weapon
+	name = "TGS 3 XENOCRUSHER tank machine gun"
+	desc = "A much better gun that shits out bullets at ridiculous speeds, don't get in its way!"
+	icon = 'icons/obj/hardpoint_modules.dmi'
+	icon_state = "m56_cupola"
+	default_ammo = /obj/item/ammo_magazine/tank/ltaaap_minigun
+	fire_sounds = list('sound/weapons/tank_minigun_loop.ogg')
+	cooldown = 0.3 SECONDS //Minimal cooldown
+	accepted_ammo = list(/obj/item/ammo_magazine/tank/towlauncher,/obj/item/ammo_magazine/tank/m56_cupola,/obj/item/ammo_magazine/tank/flamer,/obj/item/ammo_magazine/tank/tank_slauncher)
+	range_safety_check = 0
+
+/obj/item/tank_weapon/apc_cannon
+	name = "MKV-7 utility payload launcher"
+	desc = "A double barrelled cannon which can rapidly deploy utility packages to the battlefield."
+	icon = 'icons/obj/hardpoint_modules.dmi'
+	icon_state = "APC uninstalled dualcannon"
+	default_ammo = /obj/item/ammo_magazine/tank/tank_slauncher
+	cooldown = 0.7 SECONDS //Minimal cooldown
+	accepted_ammo = list(/obj/item/ammo_magazine/tank/towlauncher,/obj/item/ammo_magazine/tank/flamer,/obj/item/ammo_magazine/tank/tank_slauncher, /obj/item/ammo_magazine/tank/tank_glauncher)

```

Comment:
The default_ammo gets added to accepted_ammo so why is the same type in this list? Additionaly, why do both of these variables exist?